### PR TITLE
fix(codec): Parse numbers in arrays correctly

### PR DESCRIPTION
**Pull request by kattrali**
## Goal

Fix parsing logic for number values within arrays

## Design

Previously, parsing numbers from JSON arrays exited prematurely if the "name" value was empty. However, name is always empty when parsing arrays, and is only used when parsing JSON objects. 

So instead of exiting early to prevent parsing an invalid name, pass nil along as the name value.

## Tests

* Added a new test for adding metadata array
* Enabled some unused (??) tests for JSON parsing

## Review

- This pull request is ready for:
  - [x] Final review
- [x] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [x] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [x] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: Tests/KSCrash/KSJSONCodec_Tests.m [hunk 1]
```diff
@@ -210,24 +210,24 @@ - (void)testSerializeDeserializeArrayStringIntl
...
-//                                                  error:&error]);
-//    XCTAssertNotNil(jsonString, @"");
-//    XCTAssertNil(error, @"");
-//    XCTAssertEqualObjects(jsonString, expected, @"");
-//    id result = [BSG_KSJSONCodec decode:toData(jsonString) options:0 error:&error];
-//    XCTAssertNotNil(result, @"");
-//    XCTAssertNil(error, @"");
-//    XCTAssertEqualObjects(result, original, @"");
-//}
+- (void)testSerializeDeserializeArrayMultipleEntries
```
Comment (paulz) [inline Tests/KSCrash/KSJSONCodec_Tests.m]: Awesome job restoring commented out tests!

File: features/handled_errors.feature [hunk 2]
```diff
@@ -1,17 +1,20 @@
 Feature: Handled Errors and Exceptions
 
-Scenario: Override errorClass and message from a notifyError() callback and discard lines from stack
+Scenario: Override errorClass and message from a notifyError() callback, customize report
```
Comment (paulz) [inline features/handled_errors.feature]: proposing adding the comment to the scenario about metadata, please adjust wording as necessary.
